### PR TITLE
fixed 519

### PR DESCRIPTION
**Pull request by dougkenik**

### CODE REVIEW

**Comment by dougkenik** [comment]
@bradymadams - I believe this fixed #522 as well

File: SmartSlicePlugin/select_tool/SmartSliceSelectTool.py [hunk 1]
```diff
@@ -183,44 +184,46 @@ def _getSelectedTriangles(
...
+                    axis = selected_face.rotation_axis()
 
-            if selected_face is not None:
-                self._update_bc_node(bc_node, selected_face, axis)
+                if selected_face is not None:
+                    self._update_bc_node(bc_node, selected_face, axis)
 
-        return
+        except Exception as exc:
+            raise Exception()
```
Comment (bradymadams) [inline SmartSlicePlugin/select_tool/SmartSliceSelectTool.py:226]: Why catch an exception just to raise an empty one?

**Comment by dougkenik** [comment]
> Trying to understand what the fix is. The caught and re-raised exception doesn't seem like it would change anything. Is it the added `selectionChanged` emission?

The fix was the signal emit, yes. There is a connection in Selection.py that connects to the transformation changed of the selected SceneNode, and we needed to trigger it. emitting the selectionChanged signal does the trick

**Comment by bradymadams** [comment]
[Build for Rick](https://teton-cdn.s3-us-west-2.amazonaws.com/smartslice-cura-plugin/dist/SmartSlicePlugin-r21.0-519.23.curapackage)

**Comment by rickdalgarno** [comment]
@dougkenik The load visualization still does not go away when I click off the part.

**Review by bradymadams — APPROVED** [review]
Code change looks good to me. Tested against #522 and this did _not_ fix the issue, so let's see if it fixes #519 when Rick tests.

**Comment by rickdalgarno** [comment]
This issue might be specific to 4.9. The fix does not work in 4.9, but it does work in 4.7.

**Comment by rickdalgarno** [comment]
I saw a new commit. Is there a build ready or are there more changes coming?

**Comment by dougkenik** [comment]
> I saw a new commit. Is there a build ready or are there more changes coming?

No - my dev environment is not setup for 4.9 right now, so I'm building and testing on 4.9 install

**Comment by rickdalgarno** [comment]
Tested using SmartSlicePlugin-r21.0-519.27. Confirmed fix for #519.

The 2nd issue in #522 is still an issue, however the trigger is different. Steps:

1. Define a custom load and rotate both load wheels. 
2. Submit validation.
3. During validation, adjust the load angle slider. 
4. When the "cancel results?" popup shows, click "Cancel". The arrow direction changes each times you click Cancel.

**Review by bradymadams — APPROVED** [review]

**Comment by bradymadams** [comment]
I'm assuming Rick's last comment is approval for this PR for 519.

**Comment by rickdalgarno** [comment]
> I'm assuming Rick's last comment is approval for this PR for 519.

Correct. I approve this PR for 519 only. 522 is still an issue.

**Comment by bradymadams** [comment]
> > I'm assuming Rick's last comment is approval for this PR for 519.
> 
> Correct. I approve this PR for 519 only. 522 is still an issue.

522 was fixed in a separate PR so that's why I merged it.

**Pull request merged**